src/net_processing.cpp

```diff
@@ -721,8 +723,9 @@ class PeerManagerImpl final : public PeerManager
     bool RejectIncomingTxs(const CNode& peer) const;
 
     /** Whether we've completed initial sync yet, for determining when to turn
-      * on extra block-relay-only peers. */
-    bool m_initial_sync_finished GUARDED_BY(cs_main){false};
+      * on extra block-relay-only peers and the peer connections desirable
+      * services flags. */
+    std::atomic<bool> m_is_close_to_tip{false};
```

Comment:
There isn't a behavior change here. `StartExtraBlockRelayPeers()` can only be executed once, then it runs for the entire software lifetime. 

I don't think the base condition is different. Both action paths depend on the proximity to the tip. The extra block-relay-only peers functionality is activated when the node is close to the tip (20 blocks away). And the "desirable service flags" is modified on the opposite direction, when the node detect it is further than 288 blocks from the tip.

I think we should try to consolidate concepts that can be unified instead of continuing adding independent checks for each feature/validation we have.
Additionally, something that I had floating around my head when I did this was to transform the `m_initial_sync_finished` (now named `m_is_close_to_tip`) into a `SyncState` field with the different states the sync process could have (sync, close to tip, recent_lagging_behind, not sync) and a bool flag field to denote whether the chain is moving forward or not (if there is any in-flight blocks and the last received block is close to the node's clock time etc).